src/go/wsl-helper/pkg/dockerproxy/mungers/containers_create_linux.go

```diff
@@ -0,0 +1,377 @@
+)
+
+// For Linux (non-rancher-desktop WSL2 containers), we need to do a little more
+// work.  The mounts are defined in /containers/create, but used in
+// /containers/{id}/start instead; additionally, start may be called multiple
+// times (or /containers/{id}/restart may be called).
+// To handle this, we need to:
+// - in POST /containers/create:
+//   - on request, modify the bindings to point to temporary directories.
+//   - on response, record the container id + bind mappings.
+// - in POST /containers/{id}/start|restart|etc
+//   - on request, reconstruct the bind mappings as needed.
+//   - on response, remove those bind mappings.
+// - in DELETE /containers/{id}
+//   - remove the recording binding information
+// Note that all persisted info needs to live on disk; it's possible to run
+// containers while restarting the docker proxy (or indeed the machine).
+
+// mountDir is where we can keep our temporary mounts.
+const mountDir = "/mnt/wsl/rancher-desktop/run/docker-mounts"
+
+// contextKey is the key used to locate the bind manager in the request/response
+// context.  This only lasts for a single request/response pair.
+var contextKey = struct{}{}
+
+// bindManager manages the binding data (but does not do binding itself)
+type bindManager struct {
+	// Recorded entries, keyed by the random mount point string (the leaf name
+	// of the bind host location, as reported to dockerd).  Each entry is only
+	// used by one container; multiple entries may map to the same host path.
+	Entries map[string]bindManagerEntry `json:",omitempty"`
+
+	// Name of the file we use for persisting data.
+	statePath string
+
+	// Mutex for managing concurrency for the bindManager.
+	sync.RWMutex
+}
+
+// bindManagerEntry is one entry in the bind manager.  If all the fields are
+// empty, then the bind is incomplete (the container create failed) and it
+// should not be used.
+type bindManagerEntry struct {
+	ContainerId string
+	HostPath    string
+}
+
+func newBindManager() (*bindManager, error) {
+	statePath, err := xdg.StateFile("rancher-desktop/docker-binds.json")
+	if err != nil {
+		return nil, err
+	}
+
+	result := bindManager{
+		Entries:   make(map[string]bindManagerEntry),
+		statePath: statePath,
+	}
+	err = result.load()
+	if err != nil {
+		return nil, err
+	}
+	return &result, nil
+}
+
+// load the persisted bind manager data; this should only be called from
+// newBindManager().
+func (b *bindManager) load() error {
+	statePath := b.statePath
+	file, err := os.Open(statePath)
+	if err != nil {
+		if errors.Is(err, os.ErrNotExist) {
+			b.Entries = make(map[string]bindManagerEntry)
+			return nil
+		}
+		return fmt.Errorf("error opening state file %s: %w", statePath, err)
+	}
+	defer file.Close()
+	err = json.NewDecoder(file).Decode(b)
+	if err != nil {
+		return fmt.Errorf("error reading state file %s: %w", statePath, err)
+	}
+	b.statePath = statePath
```

Comment:
Right, we could just throw `b.Entries` into the encoder/decoder instead.  This would also mean we don't need to export it anymore.

Fixed in 5df0ca2e1a4e9b16604dae0c5be386f36ff6f56d.

Reply:
And that should have been "functions work on a different object..."